glean-core/android/src/main/java/mozilla/telemetry/glean/utils/GzipUtils.kt

```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.telemetry.glean.utils
+
+import java.io.ByteArrayInputStream
+import java.util.zip.GZIPInputStream
+
+/**
+ * Decompress the GZIP returned by the glean-core layer.
+ *
+ * @param data the gzipped [ByteArray] to decompress
+ * @return a [String] containing the uncompressed data.
+ */
+fun decompressGZIP(data: ByteArray): String {
+    val stringBuilder = StringBuilder()
+    GZIPInputStream(ByteArrayInputStream(data)).bufferedReader().useLines { lines ->
+        lines.forEach { stringBuilder.append(it) }
+    }
+    return stringBuilder.toString()
```

Comment:
You should be able to read the whole thing at once, using [`readText`](https://kotlinlang.org/docs/reference/idioms.html#java-7s-try-with-resources).

```suggestion
    val content = GZIPInputStream(ByteArrayInputStream(data)).bufferedReader(UTF_8).use { it.readText() }
    return content
```

Maybe [even like this](https://stackoverflow.com/questions/41000584/best-way-to-use-bufferedreader-in-kotlin):

```suggestion
    val content = GZIPInputStream(ByteArrayInputStream(data)).bufferedReader().use(BufferedReader::readText)
    return content
```

Reply:
That's.. way better!